File: 7kyu/descending_order.ts
// My solution
export function descendingOrder(n: number): number {
  return Number(`${n}`.split('').sort().reverse().join(''));
}

// Top solution
// export function descendingOrder(
//   n: number | null | undefined
// ): number | null | undefined {
//   return n ? parseInt(n.toString().split('').sort().reverse().join('')) : n;
// }

// Similar to mine
// export function descendingOrder(n: number): number {
//   return parseInt(n.toString().split('').sort().reverse().join(''));
// }

File: 7kyu/descending_order.py
# My solution

def descending_order(num):
    if num < 10:
        return num
    
    return int(''.join(reversed(sorted(list(str(num))))))
    
'''
Parameters: an integer
Return: an integer 
Examples:
    descending_order(0) -> 0
    descending_order(15) -> 51
    descending_order(123456789) -> 987654321
Pseudocode:
    if num is < 10 return num
    convert the num to a string
    split the string into a list of chars
    sort the list and reverse
    join the list
    convert back to int
'''

# Top solution

def Descending_Order(num):
    return int("".join(sorted(str(num), reverse=True)))